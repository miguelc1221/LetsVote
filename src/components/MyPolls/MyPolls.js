import React, { Component } from "react";
import { Header, Segment, Button } from "semantic-ui-react";
import { connect } from "react-redux";
import * as pollActions from "../../actions/polls";
import "./MyPolls.css";

class MyPolls extends Component {
  componentDidMount() {
    this.props.fetchPolls();
  }

  // renderPolls() {
  //   return this.props.fetchPolls.map(val => {});
  // }

  render() {
    console.log(this.props.polls);
    const { polls } = this.props;
    return (
      <div className="my-polls">
        <Header textAlign="center" as="h1">
          My polls
        </Header>
        <Segment.Group stacked className="my-polls__seg-group">
          {polls.map(val => {
            return (
              <Segment clearing className="my-polls__seg" key={val._id}>
                {val.question}
                <Button size="small" color="youtube" floated="right">
                  Delete
                </Button>
                <Button size="small" color="olive" floated="right">
                  View
                </Button>
              </Segment>
            );
          })}
        </Segment.Group>

      </div>
    );
  }
}

const mapStateToProps = state => ({
  polls: state.polls.polls
});

export default connect(mapStateToProps, pollActions)(MyPolls);
