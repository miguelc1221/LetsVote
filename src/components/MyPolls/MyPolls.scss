.my-polls {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 100px;
  margin-bottom: 100px;
  flex: 1;
  &__seg-container {
    background-color: #F5F5F5 !important;
    @media (max-width: 768px) {
      width: 100%;
    }
  }
  &__seg-group {
    width: 600px;
    @media (max-width: 768px) {
      width: 100%;
    }
  }
  &__seg {
    transition: ease .2s;
    cursor: pointer;
  }
  &__btn {
    display: inline;
    @media (max-width: 376px) {
      display: block;
      align-self: center;
    }
  }
}