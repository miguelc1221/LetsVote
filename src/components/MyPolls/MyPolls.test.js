import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MyPolls } from './MyPolls';

describe('MyPolls Component', () => {
  it('should render as expected', () => {
    const polls = [{ pollOne: 1, _id: 1}, { pollTwo: 2, _id: 2}]
    const component = shallow(<MyPolls polls={polls} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  })
})