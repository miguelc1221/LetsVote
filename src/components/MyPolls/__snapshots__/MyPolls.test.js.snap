exports[`MyPolls Component should render as expected 1`] = `
<div
  className="my-polls">
  <Segment
    className="my-polls__seg-container">
    <Header
      as="h1"
      textAlign="center">
      My Polls
    </Header>
    <SegmentGroup
      className="my-polls__seg-group"
      stacked={true}>
      <Segment
        className="my-polls__seg"
        clearing={true}>
        <div
          className="my-polls__btn">
          <Button
            as="button"
            color="youtube"
            floated="right"
            onClick={[Function]}
            size="small">
            Delete
          </Button>
          <Link
            replace={false}
            to="/chart/1">
            <Button
              as="button"
              color="olive"
              floated="right"
              size="small">
              Chart
            </Button>
          </Link>
          <Link
            replace={false}
            to="/vote/1">
            <Button
              as="button"
              color="blue"
              floated="right"
              size="small">
              Vote
            </Button>
          </Link>
        </div>
      </Segment>
      <Segment
        className="my-polls__seg"
        clearing={true}>
        <div
          className="my-polls__btn">
          <Button
            as="button"
            color="youtube"
            floated="right"
            onClick={[Function]}
            size="small">
            Delete
          </Button>
          <Link
            replace={false}
            to="/chart/2">
            <Button
              as="button"
              color="olive"
              floated="right"
              size="small">
              Chart
            </Button>
          </Link>
          <Link
            replace={false}
            to="/vote/2">
            <Button
              as="button"
              color="blue"
              floated="right"
              size="small">
              Vote
            </Button>
          </Link>
        </div>
      </Segment>
    </SegmentGroup>
  </Segment>
</div>
`;
